src/Paprika/Tree/KeccakOrRlp.cs

```diff
@@ -0,0 +1,38 @@
+using Paprika.Crypto;
+
+namespace Paprika.Tree;
+
+public ref struct KeccakOrRlp
+{
+    public enum Type : byte
+    {
+        Keccak = 0,
+        Rlp = 1
+    }
+
+    public Type DataType { get; }
+    public Span<byte> Data { get; }
+
+    public KeccakOrRlp(Type dataType, Span<byte> data)
+    {
+        DataType = dataType;
+        Data = data;
+    }
+
+    public static KeccakOrRlp WrapRlp(Span<byte> data)
+    {
+        var destination = new byte[32];
+
+        if (data.Length < 32)
+        {
+            destination[0] = (byte)data.Length;
+            data.CopyTo(destination[1..]);
+            return new KeccakOrRlp(Type.Rlp, destination);
+        }
```

Comment:
We have no test case that covers this branch. It seems that for Leaves it is impossible, but it's worth mentioning. I initially made a mistake of doing `return new KeccakOrRlp(Type.Rlp, data);` instead of `return new KeccakOrRlp(Type.Rlp, destination);`

Reply:
It could be the case if a path is 1 nibble only and the account is almost empty 👀 , like 0.0000....1 Eth and 0 nonce?